### ISSUE TITLE
test request restricted data

### ISSUE THREAD

**Issue by nicolasvanermen** [ices-tools-dev/esas#133]
@Carlos could you link the Bureau Waardenburg data to Ruben Fijn (r.c.fijn@buwa.nl) so I can test whether I can download their data?

**Comment by cmspinto** [comment]
Yes, that would be a good idea. 
I think @Osanna123 found a few typos that we will correct in the email(s).

**Comment by nicolasvanermen** [comment]
Has Ruben been linked as contact person to the BuWa data? And if not, who is?

**Comment by cmspinto** [comment]
No, currently there is only one user that is linked to an institution, it's you:
![image](https://user-images.githubusercontent.com/8705477/203734775-8643f467-c464-408c-bc08-cf78cff793ff.png)

Either you can insert the link between the users and the institutes or you can send me the list and I can do it.

In the case of Ruben what is his user name or email?

**Comment by nicolasvanermen** [comment]
His email is r.c.fijn@buwa.nl, but I don't think he has an account nor a username at ICES. Should he request this himself before we can continue? And who should he be contacting?

**Comment by nicolasvanermen** [comment]
Hi Carlos, can you guide me through this...? See the printscreen, this a request regarding restricted data (so BuWa data). Yet the data rights holder for this request is flagged INBO (code 202), while I guess it should be BuWa. But since it is flagged INBO, I think it is strange that the contact is not me but you. Can you tell what is going wrong here?

![image](https://user-images.githubusercontent.com/7381853/214793273-cbb09b80-ad30-405e-862e-f46b3d0f87ec.png)

**Comment by cmspinto** [comment]
Hi Nicolas, 

There was a bug in the query and all the institutes were able to view the request, no matter if there was restricted or public data, that is the reason why you could see the request! 
So in summary, it was not extraction the institutes that had restricted data, it was extracting all the institutes that were in the request, filter is now in place to only the institutes that have restricted data will be able to see the request, can you check?

![image](https://user-images.githubusercontent.com/8705477/214838952-5801471d-dab8-443e-9bba-5721aac66ae9.png)


This still needs more testing, if there is another institute uploading restricted data then we should take the opportunity to test and develop further this feature.

**Comment by nicolasvanermen** [comment]
Indeed, now I get an empty list, as expected. I sent an email so Ruben can receive an account.

![image](https://user-images.githubusercontent.com/7381853/216280733-3e4e6da8-9ef7-4501-b5d3-8f08c50547fa.png)

**Comment by nicolasvanermen** [comment]
Hello Carlos,
I received an email from Job De Jong of Bureau Waardenburg. They would like to get an export for their own data, but Ruben did not get an email to approve the use and export of their restricted data. How can we set this straight? They plan to upload data this spring so we should really try to encourage them as much as possible.
Thanks!
Nicolas

**Comment by cmspinto** [comment]
Hi Nicolas, if I found the right email of Job de Jong, he should now have access to the requests for Bureau Waardenburg:

![image](https://github.com/ices-tools-dev/esas/assets/8705477/0207f1cd-13a7-4198-9a5f-b086e5e4c672)
 
 
 
Let's try to go step by step:

Login here [https://esas.ices.dk/manage/](https://esas.ices.dk/manage/) and choose the option "[View list of your restricted data requests](https://esas.ices.dk/manage/ListDataRequests)": 

![image](https://github.com/ices-tools-dev/esas/assets/8705477/a04fe92d-d3af-4f91-9fad-5a68d9070e24)

Here he will see a list of all the request for his organization:

![image](https://github.com/ices-tools-dev/esas/assets/8705477/d9290b6b-9271-4454-809c-ada659c62630)

 
 
 
And yes, the last one was himself doing the request (so I can see I got correctly his email):

![image](https://github.com/ices-tools-dev/esas/assets/8705477/e1f69ad3-b453-4dc6-a1fa-21d3f357e75a)


I hope that I was clear and gave a good description, but any question or comments are very welcome. 

Note: Another option is to download the data of Bureau Waardenburg using the webservices. In case he authenticates using JWT, when requesting the data using the webservices, since he is the administrator of Bureau Waardenburg the data will be included in the download using the webservices.

**Comment by nicolasvanermen** [comment]
OK, thank you very much for the quick follow-up Carlos, I will forward this to Job and await his response.
As I understand this is a workaround for the bug in the workflow concerning restricted data requests, right? Good for now though, and let's try to fix this issue some time in the coming months...
Have a nice 2024!
Nicolas

**Comment by cmspinto** [comment]
Hi Nicolas, 

A good 2024 for you. 

I'm not sure what you mean by suggesting there is a bug. 
This represents the workflow of the requests made on restricted data.

**Comment by nicolasvanermen** [comment]
Well I assumed there to be a bug since Ruben mentions he never received my (test) data request, nor the request of Job... Yet Job de Jong notified that all went well... So what went wrong in first instance? Maybe Ruben does still not have an ICES account, could that be the problem?
